### ISSUE TITLE
[Feature Request]: `session.setPermissionCheckHandler` cannot handle 'default' state for Notification.permission.

### ISSUE THREAD

**Issue by Shintarot** [electron/electron#36232]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

`session.setPermissionCheckHandler` ([doc](https://www.electronjs.org/docs/latest/api/session#sessetpermissioncheckhandlerhandler)) can only represent a boolean and not the 'default' of `Notification.permission`.

There seems to be mention of this problem in [the PR](https://www.electronjs.org/docs/latest/api/session#sessetpermissioncheckhandlerhandler), but why is it not implemented?

### Proposed Solution

I think it is reasonable to have `"not set"` as a permission state, not just for `Notification.permission`.

Some of them may need to be boolean when they are finally returned to the `renderer` internally, but I think it is possible to implement it in such a way that it is treated as `false`.

Or, it would be better to separate them like `session.setNotificationPermissionCheckHandler(callback)` or `session.setPermissionCheckHandler(permission, callback)`. I also feel it is drastic.

### Alternatives Considered

今の仕様で`renderer`からの`Notification.permission`の照会に対応するには、`renderer`内で（`preload`または`webContents.executeJavaScript`で）`Notification.permission`を書き換える必要があります。

To handle `Notification.permission` queries from the `renderer` with the current spec, we need to override `Notification.permission` in the `renderer` (with `preload` or `webContents.executeJavaScript`).

**Comment by wheezard** [comment]
The problem actually lies around [here](https://github.com/electron/electron/blob/3a94634ae559e5e9cc41e26b56af95a3f916a7f9/shell/browser/electron_permission_manager.cc#L257), because `blink::mojom::PermissionStatus` enum has *three* members: `GRANTED`, `DENIED` and `ASK`, and the code here can never return `ASK`.

So, [`CheckPermissionWithDetails()`](https://github.com/electron/electron/blob/3a94634ae559e5e9cc41e26b56af95a3f916a7f9/shell/browser/electron_permission_manager.cc#L296) should return either a string or a number (i.e. 0 for denied, 1 for granted, 2 for ask) from the js function `setPermissionCheckHandler`.

I would sumbit a PR, but I have never worked with Chromium & Blink and have no idea how to convert js values into c++ values.

**Comment by Shintarot** [comment]
> The problem actually lies around [here](https://github.com/electron/electron/blob/3a94634ae559e5e9cc41e26b56af95a3f916a7f9/shell/browser/electron_permission_manager.cc#L257), because [`blink::mojom::PermissionStatus` enum](https://source.chromium.org/chromium/chromium/src/+/main:out/Debug/gen/third_party/blink/public/mojom/permissions/permission_status.mojom-shared.h;l=49?q=PermissionStatus&ss=chromium%2Fchromium%2Fsrc) has _three_ members: `GRANTED`, `DENIED` and `ASK`, and the code here never returns `ASK`.
> 
> So, [`CheckPermissionWithDetails()`](https://github.com/electron/electron/blob/3a94634ae559e5e9cc41e26b56af95a3f916a7f9/shell/browser/electron_permission_manager.cc#L296) should return either a string or a number (i.e. 0 for denied, 1 for granted, 2 for ask) from the js function `setPermissionCheckHandler`.
> 
> I would sumbit a PR, but I have never worked with Chromium & Blink and have no idea how to convert js values into c++ values.

Thank you for clarifying the cause. Then, do you think it is appropriate to label this issue as a bug?

**Comment by wheezard** [comment]
No, I don't think so. The code is working as expected, the issue is that the code (by design, it seems) doesn't use this capability of Chromium.
